### PR TITLE
[FLINK-24770] Remove error from JobStatusListener

### PR DESCRIPTION
**Pull request by zentol**
The error is not currently used anywhere and makes it harder to add more uses of the interface.

Specifically I want to extend the AdaptiveScheduler to send the status updates, but doing that in a generic fashion without having to touch all scheduler states isn't currently possible because the existence of a failure cause is an implementation detail of the satte.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java [hunk 1]
```diff
@@ -1042,8 +1041,7 @@ private boolean transitionState(JobStatus current, JobStatus newState, Throwable
                     newState,
                     error);
 
-            stateTimestamps[newState.ordinal()] = System.currentTimeMillis();
```
Comment (nicoweidner) [inline flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java:1045]: Was this line removed on purpose? Seems unrelated to removing the error

Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java:1049]: No that was an accident when transferring the changes from WIP branch.

**Review by nicoweidner — APPROVED** [review]
